nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/OperationContext.java

```diff
@@ -217,71 +238,91 @@ public void setRollback() {
     }
 
     public Map<String, Object> getVars() {
-        return vars;
+        return this;
     }
 
     /** the map API */
 
     @Override
-    public int size() {
-        return vars.size();
-    }
-
-    @Override
-    public boolean isEmpty() {
-        return vars.isEmpty();
-    }
-
-    @Override
     public boolean containsKey(Object key) {
-        return vars.containsKey(key);
-    }
-
-    @Override
-    public boolean containsValue(Object value) {
-        return vars.containsValue(value);
+        if (Constants.VAR_RUNTIME_CHAIN.equals(key)) {
+            return true;
+        }
+        return super.containsKey(key);
     }
 
     @Override
     public Object get(Object key) {
-        return vars.get(key);
+        if (Constants.VAR_RUNTIME_CHAIN.equals(key)) {
+            return this;
+        }
+        return resolve(vars.get(key));
     }
 
     @Override
     public Object put(String key, Object value) {
-        return vars.put(key, value);
+        if (Constants.VAR_RUNTIME_CHAIN.equals(key)) {
+            throw new IllegalArgumentException(Constants.VAR_RUNTIME_CHAIN + " is reserved, not writable");
+        }
+        return resolve(vars.put(key, value));
     }
 
     @Override
     public Object remove(Object key) {
-        return vars.remove(key);
+        if (Constants.VAR_RUNTIME_CHAIN.equals(key)) {
+            throw new IllegalArgumentException(Constants.VAR_RUNTIME_CHAIN + " is reserved, not writable");
+        }
+        return resolve(vars.remove(key));
     }
 
-    @Override
-    public void putAll(Map<? extends String, ?> m) {
-        vars.putAll(m);
-    }
 
     @Override
-    public void clear() {
-        vars.clear();
-    }
+    public Set<java.util.Map.Entry<String, Object>> entrySet() {
+        return new AbstractSet<Map.Entry<String,Object>>() {
```

Comment:
`java.util.Map.Entry` —> `Map.Entry` —> `Entry` is enough (already used in other places).